eth/downloader/queue.go

```diff
@@ -783,6 +783,10 @@ func (q *queue) DeliverBodies(id string, txLists [][]*types.Transaction, txListH
 		if header.WithdrawalsHash == nil {
 			// discard any withdrawals if we don't have a withdrawal hash set
 			withdrawalLists[index] = nil
+		} else if *header.WithdrawalsHash == types.EmptyRootHash {
+			// if the withdrawal hash is the emptyRootHash,
+			// we expect withdrawals to be [] instead of nil
+			withdrawalLists[index] = make([]*types.Withdrawal, 0)
```

Comment:
By placing this clause here, instead of as the last clause, the following can happen: 
- We get delivered a header with emptyRootHash withdrawalsHash, but a non-empty list of withdrawals in the body. 
-  We accept this, and just clear out the withdrawals list. 

I guess that is fine

Reply:
Also, this will _always_ create a new withdrawals-slice, even in the happy=path when the peer correctly did send us an empt list. Might be a bit over the top?

Reply:
Yeah I kinda think that might be a bit dangerous (DOS), but we already do the same thing with blocks without withdrawalHash